extensions/sockets/src/main/java/org/mule/module/socket/api/connection/tcp/AbstractTcpConnection.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.connection.tcp;
+
+import static java.lang.String.format;
+import static org.mule.module.socket.internal.SocketUtils.getAddress;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import org.mule.module.socket.api.connection.AbstractSocketConnection;
+import org.mule.module.socket.api.exceptions.UnresolvableHostException;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.lifecycle.Initialisable;
+import org.mule.runtime.core.api.lifecycle.InitialisationException;
+
+import java.net.InetSocketAddress;
+
+import javax.inject.Inject;
+
+/**
+ * Contains behaviour and attributes proper of TCP connection
+ *
+ * @since 4.0
+ */
+abstract class AbstractTcpConnection extends AbstractSocketConnection implements Initialisable
+{
+
+    @Inject
+    MuleContext muleContext;
+
+    protected final TcpProtocol protocol;
+
+    public AbstractTcpConnection(String host, int port, TcpProtocol protocol)
+    {
+        super(host, port);
+        this.protocol = protocol;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        initialiseIfNeeded(protocol, true, muleContext);
+    }
+
+    /**
+     * Creates an {@link InetSocketAddress} with the host and port information.
+     *
+     * @param host
+     * @param port
+     * @param failOnUnresolvedHost whether a {@link UnresolvableHostException} should be thrown if the address couldn't be resolved.
+     * @return an {@link InetSocketAddress} configured with the host and port received
+     */
+    protected InetSocketAddress getSocketAddress(String host, Integer port, boolean failOnUnresolvedHost)
+    {
+
+        InetSocketAddress address = getAddress(host, port);
+        if (address.isUnresolved() && failOnUnresolvedHost)
+        {
+            throw new UnresolvableHostException(format("Host '%s' could not be resolved", host));
```

Comment:
should this actually be the cause of a COnnectionException? maybe the network is not available